test/e2e/test/elasticsearch/builder.go

```diff
@@ -129,62 +125,60 @@ func (b Builder) WithNoESTopology() Builder {
 }
 
 func (b Builder) WithESMasterNodes(count int, resources corev1.ResourceRequirements) Builder {
-	return b.WithNodeSpec(estype.NodeSet{
+	return b.WithNodeSet(estype.NodeSet{
 		Name:  "master",
 		Count: int32(count),
 		Config: &commonv1beta1.Config{
 			Data: map[string]interface{}{
-				estype.NodeData:    "false",
-				nodeStoreAllowMMap: false,
+				estype.NodeData: "false",
 			},
 		},
 		PodTemplate: ESPodTemplate(resources),
 	})
 }
 
 func (b Builder) WithESDataNodes(count int, resources corev1.ResourceRequirements) Builder {
-	return b.WithNodeSpec(estype.NodeSet{
+	return b.WithNodeSet(estype.NodeSet{
 		Name:  "data",
 		Count: int32(count),
 		Config: &commonv1beta1.Config{
 			Data: map[string]interface{}{
-				estype.NodeMaster:  "false",
-				nodeStoreAllowMMap: false,
+				estype.NodeMaster: "false",
 			},
 		},
 		PodTemplate: ESPodTemplate(resources),
 	})
 }
 
 func (b Builder) WithNamedESDataNodes(count int, name string, resources corev1.ResourceRequirements) Builder {
-	return b.WithNodeSpec(estype.NodeSet{
+	return b.WithNodeSet(estype.NodeSet{
 		Name:  name,
 		Count: int32(count),
 		Config: &commonv1beta1.Config{
 			Data: map[string]interface{}{
-				estype.NodeMaster:  "false",
-				nodeStoreAllowMMap: false,
+				estype.NodeMaster: "false",
 			},
 		},
 		PodTemplate: ESPodTemplate(resources),
 	})
 }
 
 func (b Builder) WithESMasterDataNodes(count int, resources corev1.ResourceRequirements) Builder {
-	return b.WithNodeSpec(estype.NodeSet{
-		Name:  "masterdata",
-		Count: int32(count),
-		Config: &commonv1beta1.Config{
-			Data: map[string]interface{}{
-				nodeStoreAllowMMap: false,
-			},
-		},
+	return b.WithNodeSet(estype.NodeSet{
+		Name:        "masterdata",
+		Count:       int32(count),
 		PodTemplate: ESPodTemplate(resources),
 	})
 }
 
-func (b Builder) WithNodeSpec(nodeSpec estype.NodeSet) Builder {
-	b.Elasticsearch.Spec.NodeSets = append(b.Elasticsearch.Spec.NodeSets, nodeSpec)
+func (b Builder) WithNodeSet(nodeSet estype.NodeSet) Builder {
+	// Make sure the config specifies "node.store.allow_mmap: false".
+	// We disable mmap to avoid having to set the vm.max_map_count sysctl on test k8s nodes.
+	if nodeSet.Config == nil {
+		nodeSet.Config = &commonv1beta1.Config{Data: map[string]interface{}{}}
+	}
+	nodeSet.Config.Data["node.store.allow_mmap"] = false
```

Comment:
I could have left it as a constant at the top of the file, but this way I feel like all the logic and all comments are grouped at the same place.